### DISCUSSION TITLE
[Rootless] Issues when using systemd-homed

### DISCUSSION THREAD

**Discussion by thomascft** [containerd/nerdctl#2017] [Q&A] [answered]
I have a few machines running Arch Linux and one of them I decided to try out systemd-homed.  I followed the guide on https://rootlesscontaine.rs/ and have a working rootless containerd instance running on my non-homed laptop. During diagnosis I created a regular account on problematic system and everything worked fine. I creted a fresh homed and it showed the same symptoms. Below are some of the problematic commands and their respective debug outputs. 

`nerdctl run --debug-full --rm hello-world`
```
DEBU[0000] stateDir: /run/user/60121/containerd-rootless
DEBU[0000] rootless parent main: executing "/usr/bin/nsenter" with [-r/ -w/ --preserve-credentials -m -n -U -t 737 -F nerdctl run --debug-full --rm hello-world]
DEBU[0000] verification process skipped
FATA[0000] operation not permitted
```
After that I tried running the alpine image. 
`nerdctl run --debug --rm -it alpine`
Here's the resulting error: 
```
DEBU[0000] stateDir: /run/user/60121/containerd-rootless
DEBU[0000] rootless parent main: executing "/usr/bin/nsenter" with [-r/ -w/ --preserve-credentials -m -n -U -t 27415 -F nerdctl --debug-full run --rm -it alpine]
DEBU[0000] verification process skipped
FATA[0000] operation not permitted
```
I decided to try the archlinux image and it worked without any errors.

**Comment by AkihiroSuda** [comment, marked as answer]
Fixed in:
- https://github.com/containerd/nerdctl/pull/2064
